src/templates/scaffolding_template.rs

```diff
@@ -0,0 +1,52 @@
+// This file was autogenerated by some hot garbage in the `uniffi` crate.
+// Trust me, you don't want to mess with it!
+
+{% include "RustBuffer.rs" %}
+
+// For each enum declared in the IDL, we assume the caller as provided a corresponding
+// rust `enum`. We provide the traits for sending it across the FFI, which will fail to
+// compile if the provided struct has a different shape to the one declared in the IDL.
+//
+// The enum will be sent over the FFI as a u32, with values assigned according to the
+// order of items *as declared in the IDL file*. This might be different to the order
+// of items as declared in the rust code, but no harm will come from it.
+{% for e in ci.iter_enum_definitions() %}
+{% include "EnumTemplate.rs" %}
+{% endfor %}
+
+// For each record declared in the IDL, we assume the caller has provided a corresponding
+// rust `struct` with the declared fields. We provide the traits for sending it across the FFI.
+// If the caller's struct does not match the shape and types declared in the IDL then the rust
+// compiler will complain with a type error.
```

Comment:
Although the comment seems to be duplicate here as well, might be worth removing from here.